### PR TITLE
feat: poke form to create templates

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

A very simple creation flow for templates in Poké

Screenshot 2024-03-26 at 16 33 12

### CODE REVIEW

**Review by flvndvd — COMMENTED** [review]
LGTM, left a few comments.

File: front/next.config.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 const removeImports = require("next-remove-imports")();
 
 module.exports = removeImports({
-  experimental: { esmExternals: true },
+  experimental: { esmExternals: false },
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/next.config.js:5]: not entirely sure https://dust.tt/w/0ec9852c2f/assistant/ec555777ef 

tested locally + front edge

File: front/pages/poke/templates.tsx [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            return [key, value];
+          }
+          const cleanedValue = value.trim();
+          if (!cleanedValue) {
+            return null;
+          }
+          return [key, cleanedValue];
+        })
+      )
+    );
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/pages/poke/templates.tsx:178]: Hard due to typescript !

File: front/pages/poke/templates.tsx [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+              control={form.control}
+              name="presetModel"
+              title="Preset Model"
+              options={USED_MODEL_CONFIGS.map((config) => config.modelId)}
+            />
+            <SelectField
+              control={form.control}
+              name="presetTemperature"
+              title="Preset Temperature"
+              options={["deterministic", "factual", "balanced", "creative"]}
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/pages/poke/templates.tsx]: done

File: front/pages/poke/templates.tsx [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+  PokeSelectTrigger,
+  PokeSelectValue,
+} from "@app/components/poke/shadcn/ui/select";
+import { PokeTextarea } from "@app/components/poke/shadcn/ui/textarea";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { withSuperUserAuthRequirements } from "@app/lib/iam/session";
+
+export const getServerSideProps = withSuperUserAuthRequirements<object>(
+  async (context, auth) => {
+    if (!auth.isDustSuperUser()) {
```
Comment (flvndvd) [inline front/pages/poke/templates.tsx:42]: ✂️ No need for this with the wrapper!

File: types/src/front/assistant/builder.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  "balanced",
+  "creative",
+] as const;
+export type AssistantCreativityLevel =
+  (typeof ASSISTANT_CREATIVITY_LEVELS)[number];
+export const AssistantCreativityLevelCodec = ioTsEnum<AssistantCreativityLevel>(
+  ASSISTANT_CREATIVITY_LEVELS,
+  "AssistantCreativityLevel"
+);
+export const ASSISTANT_CREATIVITY_LEVEL_DISPLAY_NAMES = {
```
Comment (flvndvd) [inline types/src/front/assistant/builder.ts:15]: You could have used lodash here 😛

**Review by flvndvd — APPROVED** [review]
LGTM, left a few nits.

**Pull request merged**